sycl/test/plugins/enqueue-arg-order-buffer.cpp

```diff
@@ -0,0 +1,444 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// RUN: env SYCL_PI_TRACE=2 %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: env SYCL_PI_TRACE=2 %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/accessor.hpp>
```

Comment:
Could you please clarify why this header needs to be included explicitly?

Reply:
I add testing to validate `getLinearIndex` , which is a protected function, and make a class here that let's me call it directly. There is a comment lower in the code about that.